libmamba/src/core/curl.cpp

```diff
@@ -281,21 +281,30 @@ namespace mamba
         return val;
     }
 
-    // WARNING curl_easy_getinfo MUST have its third argument pointing to long, char*, curl_slist*
-    // or double
+    // WARNING curl_easy_getinfo MUST have its third argument pointing to long,
+    // curl_off_t, char*, double, curl_slist*, curl_certinfo*, curl_tlssessioninfo*
+    // or curl_socket_t depending on the used option.
+    // cf. each option man page for more details.
+    // https://curl.se/libcurl/c/curl_easy_getinfo.html
+
+    // Note that `curl_off_t` can be either `long` or `long long` depending on the platform.
+    // Since `long` specialization is needed for some options (e.g CURLINFO_RESPONSE_CODE),
+    // defining `long long` is needed to handle `curl_off_t` is `long long` case without
+    // causing duplication.
+
     template tl::expected<long, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<char*, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<double, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<long long, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<curl_slist*, CURLcode> CURLHandle::get_info(CURLINFO option);
 
-    template <class I>
-    tl::expected<I, CURLcode> CURLHandle::get_integer_info(CURLINFO option)
+    template <>
+    tl::expected<std::size_t, CURLcode> CURLHandle::get_info(CURLINFO option)
     {
-        auto res = get_info<long>(option);
+        auto res = get_info<curl_off_t>(option);
         if (res)
         {
-            return static_cast<I>(res.value());
+            return static_cast<std::size_t>(res.value());
```

Comment:
I maybe should've asked earlier but isn't casting to `size_t` a mistake since before it was a signed integer?

Reply:
Because it's used to set `speed` and `m_downloaded_size` which are eventually converted to `std::size_t` in `set_speed`, `update_progress` and `set_total` of the `ProgressBar`...
These notions with `m_retry_wait_seconds` are also intuitively representing something that should be unsigned... I wonder why it's not handled in `libcurl` that way btw...

Reply:
Maybe they use negative values for errors.

Reply:
If that's the case, we should check the return value and return the appropriate `tl::unexpected`.